src/collision.cpp

```diff
@@ -355,6 +290,151 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
 			cinfo.emplace_back(true, 0, p, box);
 		}
 	}
+	return any_position_valid;
+}
+
+static void addObjectBoxes(Environment *env,
+		const aabb3f &box_0, f32 dtime,
+		const v3f &pos_f, const v3f &speed_f, ActiveObject *self,
+		std::vector<NearbyCollisionInfo> &cinfo)
+{
+	/* add object boxes to cinfo */
+
+	std::vector<ActiveObject*> objects;
+#ifndef SERVER
+	ClientEnvironment *c_env = dynamic_cast<ClientEnvironment*>(env);
+	if (c_env != 0) {
+		// Calculate distance by speed, add own extent and 1.5m of tolerance
+		f32 distance = speed_f.getLength() * dtime +
+			box_0.getExtent().getLength() + 1.5f * BS;
+		std::vector<DistanceSortedActiveObject> clientobjects;
+		c_env->getActiveObjects(pos_f, distance, clientobjects);
+
+		for (auto &clientobject : clientobjects) {
+			// Do collide with everything but itself and the parent CAO
+			if (!self || (self != clientobject.obj &&
+					self != clientobject.obj->getParent())) {
+				objects.push_back((ActiveObject*) clientobject.obj);
+			}
+		}
+	}
+	else
+#endif
+	{
+		ServerEnvironment *s_env = dynamic_cast<ServerEnvironment*>(env);
+		if (s_env != NULL) {
+			// Calculate distance by speed, add own extent and 1.5m of tolerance
+			f32 distance = speed_f.getLength() * dtime +
+				box_0.getExtent().getLength() + 1.5f * BS;
+
+			// search for objects which are not us, or we are not its parent
+			// we directly use the callback to populate the result to prevent
+			// a useless result loop here
+			auto include_obj_cb = [self, &objects] (ServerActiveObject *obj) {
+				if (!obj->isGone() &&
+					(!self || (self != obj && self != obj->getParent()))) {
+					objects.push_back((ActiveObject *)obj);
+				}
+				return false;
+			};
+
+			std::vector<ServerActiveObject *> s_objects;
+			s_env->getObjectsInsideRadius(s_objects, pos_f, distance, include_obj_cb);
+		}
+	}
+
+	for (std::vector<ActiveObject*>::const_iterator iter = objects.begin();
+			iter != objects.end(); ++iter) {
+		ActiveObject *object = *iter;
+
+		if (object && object->collideWithObjects()) {
+			aabb3f object_collisionbox;
+			if (object->getCollisionBox(&object_collisionbox))
+				cinfo.emplace_back(object, 0, object_collisionbox);
+		}
+	}
+#ifndef SERVER
+	if (self && c_env) {
+		LocalPlayer *lplayer = c_env->getLocalPlayer();
+		if (lplayer->getParent() == nullptr) {
+			aabb3f lplayer_collisionbox = lplayer->getCollisionbox();
+			v3f lplayer_pos = lplayer->getPosition();
+			lplayer_collisionbox.MinEdge += lplayer_pos;
+			lplayer_collisionbox.MaxEdge += lplayer_pos;
+			ActiveObject *obj = (ActiveObject*) lplayer->getCAO();
+			cinfo.emplace_back(obj, 0, lplayer_collisionbox);
+		}
+	}
+#endif
+}
+
+collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
```

Comment:
Might it be possible to move this function above `addAreaNodeBoxes` to avoid unnecessarily changed lines? At least in the web view it's very hard to tell which lines were changed or which that were moved.